### PR TITLE
Allow boolean as default parameter

### LINKED ISSUE #1464: Bool default parameters not working
### Version Information

* vyper Version (output of `vyper --version`): 0.1.0b10
* OS: linux
* Python Version (output of `python --version`): 3.6.7
* Environment (output of `pip freeze`): (At end of report for readability)

### What's your issue about?

* full output of the error: Default parameter values have to be literals.
* what command you ran: vyper test.vy
* the code that caused the failure:
test.vy is the following:
```
@public
def test(a: bool = True) -> bool:
    return a
```

### How can it be fixed?

Not sure precisely how to fix this, but this did work in 0.1.0b09, indicating some code change caused this.

#### Environment
apturl==0.5.2
asn1crypto==0.24.0
Brlapi==0.6.6
certifi==2018.1.18
chardet==3.0.4
chrome-gnome-shell==0.0.0
command-not-found==0.3
cryptography==2.1.4
cupshelpers==1.0
defer==1.0.6
devedeng==4.14.0
distro-info===0.18ubuntu0.18.04.1
httplib2==0.9.2
idna==2.6
keyring==10.6.0
keyrings.alt==3.0
language-selector==0.1
launchpadlib==1.10.6
lazr.restfulclient==0.13.5
lazr.uri==1.0.3
louis==3.5.0
macaroonbakery==1.1.3
Mako==1.0.7
MarkupSafe==1.0
netifaces==0.10.4
oauth==1.0.1
olefile==0.45.1
pexpect==4.2.1
Pillow==5.1.0
protobuf==3.0.0
pycairo==1.16.2
pycrypto==2.6.1
pycryptodome==3.8.2
pycups==1.9.73
pygobject==3.26.1
pymacaroons==0.13.0
PyNaCl==1.1.2
pyRFC3339==1.0
python-apt==1.6.4
python-dateutil==2.6.1
python-debian==0.1.32
pytz==2018.3
pyxattr==0.6.0
pyxdg==0.25
PyYAML==3.12
reportlab==3.4.0
requests==2.18.4
requests-unixsocket==0.1.5
SecretStorage==2.3.1
simplejson==3.13.2
six==1.11.0
system-service==0.3
systemd-python==234
ubuntu-drivers-common==0.0.0
ufw==0.36
unattended-upgrades==0.1
urllib3==1.22
usb-creator==0.3.3
virtualenv==15.1.0
vyper==0.1.0b10
wadllib==1.3.2
xkit==0.0.0
youtube-dl==2018.3.14
zope.interface==4.3.2

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
Fix #1464 

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/signatures/function_signature.py [hunk 1]
```diff
@@ -168,7 +168,8 @@ def from_definition(cls,
 
         # Validate default values.
         for default_value in getattr(code.args, 'defaults', []):
-            if not isinstance(default_value, (ast.Num, ast.Str, ast.Bytes, ast.List)):
+            allowed_types = (ast.Num, ast.Str, ast.Bytes, ast.List, ast.NameConstant)
```
Comment (jacqueswww) [inline vyper/signatures/function_signature.py]: I think we have is_literal on the types directly now, wold be properly clearer to use that ;)

**Comment by jacqueswww** [comment]
Add test and change and this can be merged :)

**Comment by charles-cooper** [comment]
@jacqueswww I agree that using is_literal is more natural, but the values for defaults don't get parsed until much later (`parse_[private,public]_function.py`), I am hesitant to move the type checker code to that stage, and I also don't want to do extra calls to parse to LLL since there may be unintended side-effects. Maybe I can introduce a function for determining if an AST node is a literal?

**Comment by charles-cooper** [comment]
It's also a little awkward, because I don't think the current check descends into the AST node anyways. For example this compiles:
```python
x: int128

@public
def test(a: bool = True) -> bool:
    return a

@public
def test2(a: int128[3] = [1, self.x, 3]):
    pass
```

**Comment by jacqueswww** [comment]
@charles-cooper you are right, let's leave as is.

**Pull request merged**